### PR TITLE
Describe resolver hierarchies and unions in How to add a GraphQL query

### PR DESCRIPTION
**Pull request by cbart**
Part of #44683

While working on a change to the GraphQL API, it took a while to figure out how resolvers interact, whether type names or method names are relevant for resolution, and how to support returning unions. Maybe this helps the next person.

## Test plan

Reviewing the documentation change.

### CODE REVIEW

File: doc/dev/how-to/add_graphql_query.md [hunk 1]
```diff
@@ -26,6 +26,90 @@ The GraphQL schema also needs to know about this new function and what it return
 newQuery(someArg: String!): Boolean!
 ```
 
+### Resolvers hierarchy
+
+GraphQL queries can be deeply nested. This means that a GraphQL request text specifies which fields of the schema should be returned. On the back-end this is reflected by a hierarchy of resolvers that stem from `schemaResolver`. Each method represent a nesting layer and returns the resolver for the nested structure.
```
Comment (sashaostrikov) [inline doc/dev/how-to/add_graphql_query.md:31]: nit: typo

```suggestion
GraphQL queries can be deeply nested. This means that a GraphQL request text specifies which fields of the schema should be returned. On the back-end this is reflected by a hierarchy of resolvers that stem from `schemaResolver`. Each method represents a nesting layer and returns the resolver for the nested structure.
```

File: doc/dev/how-to/add_graphql_query.md [hunk 2]
```diff
@@ -26,6 +26,90 @@ The GraphQL schema also needs to know about this new function and what it return
...
+A feature flag that has a statically configured value
+"""
+type FeatureFlagBoolean { ... }
+
+"""
+A feature flag that is randomly evaluated to a boolean based on the rollout parameter
+"""
+type FeatureFlagRollout { ... }
+```
+
```
Comment (sashaostrikov) [inline doc/dev/how-to/add_graphql_query.md:112]: probably a link to the GraphQL doc can be added "for further reading", wdyt?

https://graphql.org/learn/schema/#union-types

**Review by sashaostrikov — APPROVED** [review]
LGTM! Thanks for adding this doc!

**Review by ryanslade — APPROVED** [review]
Nice, thanks for adding this!

**Review by mrnugget — APPROVED** [review]

File: doc/dev/how-to/add_graphql_query.md (hunk 2 above)
Comment (cbart) [reply doc/dev/how-to/add_graphql_query.md:112]: Thanks!

**Pull request merged**